### PR TITLE
AutoML as a module

### PR DESCRIPTION
**Pull request by michalkurka**
This PR converts AutoML into an H2O extension module and removes it from regular H2O build. User will have to download H2O and AutoML extension separately.

Starting from command line:
1. Download h2o.jar and h2o-automl.jar, put h2o-automl.jar in directory h2o-ext.
2. Run java -jar h2o.jar as usual, extension will be automatically picked up by H2O.

Starting from R/Python:
Put h2o-automl.jar on classpath when running h2o.init().
- in R: h2o.init(extra_classpath="/Users/mkurka/git/h2o/h2o-3/build/h2o-ext/h2o-automl.jar")
- in Python: h2o.init(extra_classpath=['/Users/mkurka/git/h2o/h2o-3/build/h2o-ext/h2o-automl.jar'])

Starting on Hadoop:
hadoop jar h2odriver.jar -libjars /Users/mkurka/git/h2o/h2o-3/build/h2o-ext/h2o-automl.jar -nodes 1 -mapperXmx 4g -output t2

Note: Flow will show a generic error when AutoML is triggered, py/r will should descriptive error.

### CODE REVIEW

**Comment by michalkurka** [comment]
@h2o-ops please test!

**Comment by michalkurka** [comment]
@h2o-ops please test!

**Comment by michalkurka** [comment]
@h2o-ops please test!

**Comment by michalkurka** [comment]
@h2o-ops please test!

**Comment by michalkurka** [comment]
@h2o-ops please test!

**Comment by michalkurka** [comment]
@h2o-ops please test!

**Comment by michalkurka** [comment]
@h2o-ops please test!

**Comment by michalkurka** [comment]
@h2o-ops please test!

**Comment by michalkurka** [comment]
@h2o-ops please test!

**Comment by michalkurka** [comment]
@h2o-ops please test!

**Comment by michalkurka** [comment]
@h2o-ops please test!

File: h2o-r/h2o-package/R/automl.R [hunk 1]
```diff
@@ -210,3 +202,19 @@ predict.H2OAutoML <- function(object, newdata, ...) {
...
+#' @export
+h2o.automl.available <- function() {
+  "AutoML" %in% h2o.list_api_extensions()
+}
+
+.h2o.automl.na.error <- function() {
+  message("
+         ******************************************************************************************\n
+         * Please verify that you have the AutoML extension!                                      *\n
+         * Make sure the h2o-automl.jar is stored in h2o-ext directory (relative to your h2o.jar) *\n
```
Comment (mmalohlava) [inline h2o-r/h2o-package/R/automl.R]: We should put here description how to download AutoML extension.

Comment (michalkurka) [inline h2o-r/h2o-package/R/automl.R]: Good point, I think it should ideally point to documentation for getting AutoML and starting on different platforms.

**Pull request closed without merging**